02_oop_part_1/code/src/Inventory.ts

```diff
@@ -0,0 +1,22 @@
+import { Item } from './Item';
+import { ItemComparator } from './ItemComparator';
+
+export class Inventory {
+    items: Item[] = [];
+
+    addItem(item: Item) {
+        this.items.push(item)
+    }
+
+    sort(comparator?: ItemComparator): void {

```

Comment:
I believe, in this case, it has to be function overload

Inventory.sort(): This sorts the items in the Inventory instance based on their value
Inventory.sort(comparator: ItemComparator): This sorts the items in the Inventory instance based on their weight.

Reply:
yeah but as we agreed on previous Q&A that it is similar way of implementation, so I just used optional param. Is is critical?

Reply:
done :)

Reply:
it works the same, but there is an requirement in the task ;)